cranelift/unwinder/build.rs

```diff
@@ -0,0 +1,19 @@
+fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
+
+    // Determine if the current host architecture is supported by Cranelift
+    // meaning that we might be executing native code.
+    let has_host_compiler_backend = match std::env::var("CARGO_CFG_TARGET_ARCH").unwrap().as_str() {
+        "x86_64" | "riscv64" | "s390x" | "aarch64" => true,
+        _ => false,
+    };
```

Comment:
Ideally I'd prefer to not duplicate this with the main Wasmtime crate as I suspect the two will diverge over time. For example on iOS we should probably disable cranelift by default despite having a backend for aarch64. Now that doesn't actually matter for this crate though since it'll still all compile just fine on iOS (in theory), it just wouldn't be useful to have at runtime.

Would it be possible to actually drop this build script entirely? For example the `arch` module would become naturally empty on unsupported architectures and structures like `UnwindHost` would only appear for supported architectures. That'd mean that this build script would basically fall out of the `cfg-if` if/else branches for the arch support

Reply:
Done as suggested in Cranelift meeting today -- there are now `pub use`s to re-export with doc-comments attached. `cfg`s minimized and put right next to each other -- only the one `cfg_if` chain for the implementations and one with `cfg(any(...))` to wrap all the `pub use`s and the `UnwindHost` impl.

I've verified that `wasmtime-unwinder` checks on `i686-unknown-linux-gnu`.